src/mlpack/core/cv/k_fold_cv_impl.hpp

```diff
@@ -297,8 +365,13 @@ arma::Mat<ElementType> KFoldCV<MLAlgorithm,
     arma::Mat<ElementType>& m,
     const size_t i)
 {
-  return arma::Mat<ElementType>(m.colptr(binSize * i), m.n_rows,
-      trainingSubsetSize, false, true);
+  // If this is the last fold, we have to handle it a little bit differently,
+  // since the last fold may not contain 'binSize' points.
+  const size_t subsetSize = (i == k - 1) ? lastBinSize + (k - 2) * binSize :
+      trainingSubsetSize;
```

Comment:
If I understand it right, the whole story with `lastBinSize` is to make use of all data when the number of the samples is not divisible by `k` (the previous code will miss some samples during each run in this case). If it's true, then I guess we need to use this "last bin" in a training subset for `i` from `1` to `k - 1` rather than just when `i == k - 1`.  We probably also should get rid of the variable `trainingSubsetSize`, since with this implementation it makes the code more confusing.

Reply:
By reading your initial message for this PR I see that we are probably not on same page about the last cross-validation bin. I think that when we do something like this
```
binSize = source.n_cols / k;
trainingSubsetSize = binSize * (k - 1);
lastBinSize = source.n_cols - ((k - 1) * binSize);
```
`lastBinSize` is actually no less than `binSize`.

Reply:
Since `source.n_cols / k` is integer math it's effectively `floor(source.n_cols / k)`, so actually I had thought about this a little wrong.  I had been thinking `lastBinSize` would be less than `binSize` but it will actually be up to `k` larger than `binSize`.  But I think that is no problem.

I agree that `trainingSubsetSize` is now confusing since it can change, so I've removed the variable and replaced it with the manual calculation of `binSize * (k - 1)` or `binSize * (k - 2) + lastBinSize`.

Reply:
You're right, thanks for the clarification.